rest/config_legacy.go

```diff
@@ -236,6 +251,62 @@ func (lc *LegacyServerConfig) ToStartupConfig() (*StartupConfig, DbConfigMap, er
 	return &sc, dbs, nil
 }
 
+func gocbv1tov2AddressUpgrade(server string) (newServer, username, password string, err error) {
+	connSpec, err := connstr.Parse(server)
+	if err != nil {
+		return "", "", "", err
+	}
+
+	splitServers := strings.Split(server, ",")
+	// Only first host is supported to have username and pass or else gocbconnstr.Parse does not work correctly
+	if strings.Contains(splitServers[0], "@") {
+		u, err := url.Parse(splitServers[0])
+		if err != nil {
+			return "", "", "", err
+		}
+		urlUsername := u.User.Username()
+		urlPassword, urlPasswordSet := u.User.Password()
+		if urlUsername != "" && urlPasswordSet {
+			username = urlUsername
+			password = urlPassword
+		}
+	}
+
+	var hosts string
+	for _, addr := range connSpec.Addresses {
+		if connSpec.Scheme == "http" {
+			if addr.Port != 8091 {
+				return "", "", "", fmt.Errorf("automatic connection string conversion does not support non-default host ports. " +
+					"Please change the server field to use the couchbase(s):// scheme")
+			}
+			addr.Port = -1
+		}
+
+		var port string
+		if addr.Port != -1 {
+			port = ":" + strconv.Itoa(addr.Port)
+		}
+
+		hosts = hosts + addr.Host + port + ","
+	}
+
+	if connSpec.Scheme == "http" {
+		connSpec.Scheme = "couchbase"
+	}
+
+	server = connSpec.Scheme + "://" + hosts
```

Comment:
connstr.ConnSpec looks like it's already got a String() function - what's the reason to build up our own server string, instead of building a ConnSpec and using the existing String()?